chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -370,7 +370,15 @@ abstract class Record extends Aggregate {
     elements.toIndexedSeq.reverse.map(e => eltPort(e._2)).mkString("{", ", ", "}")
   }
 
-  private[chisel3] lazy val flatten = elements.toIndexedSeq.flatMap(_._2.flatten)
+  private[core] override def typeEquivalent(that: Data): Boolean = that match {
+    case that: Record =>
+      this.getClass == that.getClass &&
```

Comment:
I think there's potential debate to be had here. I think a stricter check makes sense, since I think there's an argument to be made for type equality having meaning.